src/lib/bootstrap_controller/bootstrap_controller.ml

```diff
@@ -549,7 +545,6 @@ let run_cycle ~context:(module Context : CONTEXT) ~trust_system ~verifier
           [%log error]
             ~metadata:[ ("error", Error_json.error_to_yojson e) ]
             "Local state sync failed: $error. Retry bootstrap" ;
-          Writer.close sync_ledger_writer ;
```

Comment:
Who is in charge of closing this pipe? I suggest we document on the pipe creation if the closing is done by an upperstream swappable pipe